rust/crates/langsmith-pyo3/src/py_run.rs

```diff
@@ -278,3 +331,57 @@ fn extract_value(value: &Bound<'_, PyAny>) -> PyResult<sonic_rs::Value> {
         unreachable!("failed to convert python data {value} to sonic_rs::Value")
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::test_infra::with_python_interpreter;
+    use pyo3::{prelude::*, types::PyDict};
+
+    #[pyfunction]
+    fn extract_uuid(uuid_value: &Bound<'_, PyAny>, string_value: &str) {
+        let extracted = super::extract_string_like(uuid_value).expect("extraction failed");
+        assert_eq!(extracted.as_str(), string_value);
+    }
+
+    #[test]
+    fn test_uuid_value_extraction() {
+        fn inner(py: Python<'_>) -> PyResult<()> {
+            // This call only works correctly "the first time".
+            // We use `cargo-nextest` to ensure we run each test in its own process.
+            // Otherwise, tests will suffer unpredictable errors.
+            orjson::init_typerefs();
+
+            // Create a new test module.
+            let test_module = PyModule::new_bound(py, "test_module")?;
+            test_module.add_function(pyo3::wrap_pyfunction!(extract_uuid, &test_module)?)?;
+
+            // Get `sys.modules`, then insert our module into it.
+            let sys = PyModule::import_bound(py, "sys")?;
+            let py_modules: Bound<'_, PyDict> = sys.getattr("modules")?.downcast_into()?;
+            py_modules.set_item("test_module", test_module)?;
+
+            // Now we can import and run our python code.
+            let python_code = "\
+import uuid
+import test_module
+
+uuid_to_test = uuid.uuid4()
+
+test_module.extract_uuid(uuid_to_test, str(uuid_to_test))
+            ";
+            Python::run_bound(py, python_code, None, None)?;
+
+            Ok(())
+        }
+
+        with_python_interpreter(inner).expect("encountered an unexpected error")
+    }
```

Comment:
I can make this whole thing more elegant with a bit more effort. I didn't want to commit that time right away unless you think it's a good idea to polish up the test suite some more.